AppController/djinn.rb

```diff
@@ -4287,24 +4227,26 @@ def write_locations
     login_ip = @options['login']
     master_ips = []
     memcache_ips = []
-    num_of_nodes = @nodes.length.to_s
     search_ips = []
     slave_ips = []
     taskqueue_ips = []
-
     my_public = my_node.public_ip
     my_private = my_node.private_ip
 
     # Populate the appropriate list.
-    @nodes.each { |node|
-      all_ips << node.private_ip
-      load_balancer_ips << node.private_ip if node.is_load_balancer?
-      master_ips << node.private_ip if node.is_db_master?
-      memcache_ips << node.private_ip if node.is_memcache?
-      search_ips << node.private_ip if node.is_search?
-      slave_ips << node.private_ip if node.is_db_slave?
-      taskqueue_ips << node.private_ip if node.is_taskqueue_master? ||
-        node.is_taskqueue_slave?
+    num_of_nodes = 0
+    @state_change_lock.synchronize {
+      num_of_nodes = @nodes.length.to_s
```

Comment:
because it will be used a string, so I thought it was clear to make it a string here.